src/tokenizedSubaccounts/PPTSA.sol

```diff
@@ -0,0 +1,578 @@
+  ///      1       |       0      |    1    |   0
+  ///      1       |       1      |    0    |   0
+  ///      1       |       1      |    1    |   1
+  function _verifyHigherStrikeAmount(int strikeAmount, bool isTaker) internal pure {
+    PPTSAStorage memory $ = _getPPTSAStorage();
+    bool makerShouldBeSellingSpread = ($.isCallSpread == $.isLongSpread) ? isTaker : !isTaker;
+    if (makerShouldBeSellingSpread && strikeAmount <= 0) {
+      revert PPT_InvalidHighStrikeAmount();
+    } else if (!makerShouldBeSellingSpread && strikeAmount >= 0) {
+      revert PPT_InvalidHighStrikeAmount();
+    }
+  }
+
+  function _validateAndSortStrikes(IRfqModule.TradeData[] memory makerTrades)
+    internal
+    view
+    returns (StrikeData memory lowerStrike, StrikeData memory higherStrike)
+  {
+    StrikeData memory strike1 = _validateAndCreateIndividualStrike(makerTrades[0]);
+    StrikeData memory strike2 = _validateAndCreateIndividualStrike(makerTrades[1]);
+
+    if (
+      makerTrades[0].asset != makerTrades[1].asset || makerTrades[0].asset != address(_getPPTSAStorage().optionAsset)
+        || strike1.expiry != strike2.expiry
+    ) {
+      revert PPT_InvalidParams();
+    }
+
+    if (strike1.strike > strike2.strike) {
+      return (strike2, strike1);
+    }
+    return (strike1, strike2);
+  }
+
+  function _validateAndCreateIndividualStrike(IRfqModule.TradeData memory trade)
+    internal
+    view
+    returns (StrikeData memory)
+  {
+    PPTSAStorage storage $ = _getPPTSAStorage();
+    (uint optionExpiry, uint optionStrike, bool isCall) = OptionEncoding.fromSubId(trade.subId.toUint96());
+
+    if (
+      optionExpiry < block.timestamp + $.ppParams.optionMinTimeToExpiry
+        || optionExpiry > block.timestamp + $.ppParams.optionMaxTimeToExpiry
+    ) {
+      revert PPT_OptionExpiryOutOfBounds();
+    }
+
+    (uint callPrice, uint putPrice,) = _getOptionPrice(optionExpiry, optionStrike);
+    uint markPrice = isCall ? callPrice : putPrice;
+    if ($.isCallSpread != isCall) {
+      revert PPT_WrongInputSpread();
+    }
+    return StrikeData({
+      strike: optionStrike,
+      expiry: optionExpiry,
+      markPrice: markPrice.toInt256(),
+      tradePrice: trade.price,
+      tradeAmount: trade.amount
+    });
+  }
+
+  function _verifyRfqExecute(StrikeData memory lowerStrike, StrikeData memory higherStrike, uint maxFee) internal view {
+    PPTSAStorage storage $ = _getPPTSAStorage();
+    uint strikeDiff = higherStrike.strike - lowerStrike.strike;
+    if (strikeDiff != _getPPTSAStorage().ppParams.strikeDiff) {
+      revert PPT_StrikePriceOutsideOfDiff();
+    }
+
+    // we always require the trade to be one sell and one buy of equal size, although prices can be different.
+    if (higherStrike.tradeAmount != -lowerStrike.tradeAmount) {
+      revert PPT_InvalidTradeAmount();
+    }
+
+    (uint openSpreads, uint baseBalance,) = _getSubAccountStats();
+
+    _verifyFee(maxFee, _getBasePrice());
```

Comment:
This needs to be different to the way fees are calculated for collateral. 

Would rename to `_verifyCollateralTradeFee` and create a new function `_verifyRFQFee`. 

The formula would need a new param called `rfqFeeFactor` and formula would look like this:

if `maxFee > higherStrike.tradeAmount.abs() + lowerStrike.tradeAmount.abs() * rfqFeeFactor * _getBasePrice()` -> revert

Obviously the above can be cleaner depending on how you create `_verifyRFQFee`

Reply:
Added the new param and talked it over.